opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -2365,33 +2381,202 @@ namespace Opm
     }
 
 
+    template<typename TypeTag>
+    void
+    MultisegmentWell<TypeTag>::
+    checkOperabilityUnderBHPLimitProducer(const WellState& well_state, const Simulator& ebos_simulator, Opm::DeferredLogger& deferred_logger)
+    {
+        const auto& summaryState = ebos_simulator.vanguard().summaryState();
+        const double bhp_limit = Base::mostStrictBhpFromBhpLimits(summaryState);
+        // Crude but works: default is one atmosphere.
+        // TODO: a better way to detect whether the BHP is defaulted or not
+        const bool bhp_limit_not_defaulted = bhp_limit > 1.5 * unit::barsa;
+        if ( bhp_limit_not_defaulted || !this->wellHasTHPConstraints(summaryState) ) {
+            // if the BHP limit is not defaulted or the well does not have a THP limit
+            // we need to check the BHP limit
+
+            double temp = 0;
+            for (int p = 0; p < number_of_phases_; ++p) {
+                temp += ipr_a_[p] - ipr_b_[p] * bhp_limit;
+            }
+            if (temp < 0.) {
+                this->operability_status_.operable_under_only_bhp_limit = false;
+            }
 
```

Comment:
@GitPaean For the standard well every phase is checked. For phases with very small rates this could lead to immature shutting of wells. What do you think? Should I change this for the standard wells as well?

Reply:
The well operability formulation is really far from ideal. Both Flow and commercial simulator can shut well prematurely because of the simplified model.  For this sense, if algorithm can reduce the shutting of the wells in general, it should be okay. 

> For phases with very small rates this could lead to immature shutting of wells. 

I think here you mean it shuts well less. Ideally, check three phases individually and check the sum should be the same. I am okay if you want to check the sum to reduce the shutting of the wells because of the reason stated above. Maybe can be slightly better, if you check the weighted sum with the scaling factors (similar to the weighted total rate), since the purpose is to avoid small rate values causing difficulty in determining the sign. Weighting might help to avoid some other problems when gas rate is dominating. 

Maybe the change related to StandardWell can be done in a separate PR for better testing.